src/app/campaign-list/campaign-list.service.ts

```diff
@@ -232,4 +246,60 @@ export class CampaignListService {
     }
     return filters;
   }
+
```

Comment:
I pulled this code from the campaign-list.component to re-use it for building the queryParams